076-support-for-kafka-4.0.md

```diff
@@ -0,0 +1,203 @@
+# Support for Apache Kafka 4.0
+
+Apache Kafka is planning a new major release - Apache Kafka 4.0.
+Unlike previous major releases that Strimzi handled before (2.0 and 3.0), this one is "special" because it drops the support for ZooKeeper-based clusters.
+This proposal discusses how Strimzi will adopt the release of Apache Kafka 4.0 and transition from Apache Kafka 3.x to 4.x.
+
+## Current situation
+
+### Strimzi
+
+Strimzi currently does a new minor release every roughly every 2 months.
+Each release supports at least the two last Kafka minor versions.
+For example 3.6.x and 3.7.x.
+For each minor version, multiple patch releases might be supported, depending on their availability in Apache Kafka.
+When a new minor Apache Kafka version is released, Strimzi will typically adopt the new minor version and drop support for the oldest supported minor Kafka version.
+Typically, this happens every other Strimzi release as Apache Kafka has slower release cycles compared to Strimzi.
+But in some cases, there might be more Strimzi releases in between (for example, Strimzi 0.40, 0.41 and 0.42 are all based on Kafka 3.6.x and 3.7.x).
+
+Once a new minor Strimzi version is released, we will stop supporting the previous Strimzi release.
+There are some exceptions, especially in the following cases:
+* Serious bugs or CVEs
+* The latest minor Strimzi release is new and some people might not have had chance to upgrade to it
+
+#### KRaft support
+
+Strimzi currently supports both ZooKeeper- and KRaft-based Apache Kafka clusters.
+It also supports migration of ZooKeeper-based clusters to KRaft.
+The current limitations to the KRaft support are based on limitations in Apache Kafka itself.
+At the time of writing this proposal, limitations include:
+* Scaling of KRaft controller nodes up or down is not supported
+* Unregistering Kafka nodes removed from the Kafka cluster
```

Comment:
In case some users read this proposal but doesn't have context what we are talking about here, we could add KIP/JIRA number:

```
* Scaling of KRaft controller nodes up or down is not supported (KIP-853)
* Unregistering Kafka nodes removed from the Kafka cluster (KAFKA-17094)
```